src/Shared/UnitTests/Vacancies.Client/Application/CommandHandlers/PatchVacancyTrainingProviderCommandHandlerTests.cs

```diff
@@ -0,0 +1,116 @@
+using FluentAssertions;
+using Xunit;
+using Moq;
+using Microsoft.Extensions.Logging;
+using Esfa.Recruit.Vacancies.Client.Application.CommandHandlers;
+using Esfa.Recruit.Vacancies.Client.Domain.Repositories;
+using Esfa.Recruit.Vacancies.Client.Domain.Messaging;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Application.Commands;
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Services.TrainingProvider;
+
+namespace Esfa.Recruit.UnitTests.Vacancies.Client.Application.CommandHandlers
+{
+    [Trait("Category", "Unit")]
+    public class PatchVacancyTrainingProviderCommandHandlerTests
+    {
+        private const long EsfaUkprn = 10033670;
+        private const long MockedProviderUkprn = 12345678;
+        private readonly Mock<IVacancyRepository> _mockRepository;
+        private readonly Mock<ITrainingProviderService> _mockTrainingProvider;
+        private readonly PatchVacancyTrainingProviderCommandHandler _handler;
+        private Vacancy _updatedVacancy = null;
+
+        public PatchVacancyTrainingProviderCommandHandlerTests()
+        {
+            _mockRepository = new Mock<IVacancyRepository>();
+            _mockTrainingProvider = new Mock<ITrainingProviderService>();
+            
+            _mockRepository.Setup(x => x.UpdateAsync(It.IsAny<Vacancy>()))
+                            .Callback<Vacancy>(arg => _updatedVacancy = arg)
+                            .Returns(Task.CompletedTask);
+            _mockTrainingProvider.Setup(x => x.GetProviderAsync(MockedProviderUkprn))
+                                .ReturnsAsync(new TrainingProvider() { Ukprn = MockedProviderUkprn, Name = "FakeProvider", Address = new Address() })
+                                .Verifiable();
```

Comment:
There is a very good note on stackoverflow about Verifiable and why it may not be a good practice. https://stackoverflow.com/a/1728496/3917716

Reply:
I don't understand that post but I will remove it.